app/views/users/new.html.erb

```diff
@@ -1,45 +1,47 @@
-<div id='clearance'>
-  
-<div class="float-left">
-  <div class="welcome-text-container">
-    <h2 class="welcome-title">Welcome to Acebook</h2>
-    <ul class="slides">
-      <li class="slide showing">"It's the greatest social network. It has the best posts.
-        The greatest photos. It's gonna be yee-uuuuge."</li>
-      <li class="slide">"My favourite social network, it is!"</li>
-      <li class="slide">"I met my dog walker on Acebook!"</li>
-      <li class="slide">"Hmm, interesting."</li>
-      <li class="slide">"Finally, another way to feel socially inadequate!"</li>
-    </ul>
-  </div>
-</div>
-
-<div class="sign-up-form">
-  <%= form_for @user, html: {class: 'new-user'} do |form| %>
+<body class='login-page'>
```

Comment:
It's a nice idea — pretty interesting that it works — but it's not valid HTML and it only works out because when browsers encounter invalid HTML they try to guess what the right thing to do is. This falls down in some situations — e.g. if anyone wants to add a class to the body in `application.html.erb` it will behave strangely.

There are a couple ways around this — you could change this to a `div` and then use a clearfix tag at the end of it, then shift around the CSS to remove some of the margin that will cause trouble.

Or you could remove the `body` and use `content_for`, e.g.

```ruby
# app/views/users/new.html.erb
<% content_for :body_class, "login-page" %>
```

```ruby
# app/views/layouts/application.html.erb
<body class="<%= content_for :body_class %>">
```

Check the docs for more information on `content_for`. Don't overuse it though!